common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -534,8 +534,10 @@ public static UTF8String exec(
       CollationFactory.Collation collation = CollationFactory.fetchCollation(collationId);
       if (collation.supportsBinaryEquality) {
         return execBinary(srcString);
-      } else {
+      } else if (collation.supportsLowercaseEquality) {
         return execLowercase(srcString);
+      } else {
+        return execLowercase(srcString); // TODO: ICU implementation
```

Comment:
Let's file a JIRA and tag it together with TODO. e.g., `TODO(SPARK-XXXXX): ICU implementation`

Reply:
will do, this is only experimental for now